user-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/user/kafka/KafkaContractTest.java

```diff
@@ -0,0 +1,353 @@
+                () -> assertThat(results.get(0).value().get("user").get("userId").get("id").asText())
+                        .isEqualTo(ContractTestData.USER_ID.getId()),
+                () -> assertThat(results.get(0).value().get("user").get("auths").get(0).asText())
+                        .isEqualTo("auth"),
+                () -> assertThat(results.get(0).value().get("user").get("roles").get(0).asText())
+                        .isEqualTo("role")
+        );
+    }
+
+    @Test
+    @DirtiesContext
+    void testRestEndpoint() {
+        // Given - we are already listening to the service input
+        ConsumerSettings<String, UserRequest> consumerSettings = ConsumerSettings
+                .create(akkaActorSystem, SerDesConfig.requestKeyDeserializer(), SerDesConfig.requestValueDeserializer())
+                .withGroupId("test-group")
+                .withBootstrapServers(KafkaInitializer.kafka.getBootstrapServers())
+                .withProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        Probe<ConsumerRecord<String, UserRequest>> probe = Consumer
+                .atMostOnceSource(consumerSettings, Subscriptions.topics(consumerTopicConfiguration.getTopics().get("input-topic").getName()))
+                .runWith(TestSink.probe(akkaActorSystem), akkaMaterializer);
+
+        // When - we POST to the rest endpoint
+        Map<String, List<String>> headers = Collections.singletonMap(Token.HEADER, Collections.singletonList(ContractTestData.REQUEST_TOKEN));
+        HttpEntity<UserRequest> entity = new HttpEntity<>(ContractTestData.REQUEST_OBJ, new LinkedMultiValueMap<>(headers));
+        ResponseEntity<Void> response = restTemplate.postForEntity("/api/user", entity, Void.class);
+
+        // Then - the REST request was accepted
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.ACCEPTED);
+
+        // When - results are pulled from the output stream
+        Probe<ConsumerRecord<String, UserRequest>> resultSeq = probe.request(1);
+        LinkedList<ConsumerRecord<String, UserRequest>> results = LongStream.range(0, 1)
+                .mapToObj(i -> resultSeq.expectNext(new FiniteDuration(20, TimeUnit.SECONDS)))
+                .collect(Collectors.toCollection(LinkedList::new));
+
+        // Then - the results are as expected
+        // The request was written with the correct header
+        assertAll("Records returned are correct",
+                () -> assertThat(results)
+                        .hasSize(1),
+
+                () -> assertThat(results)
+                        .allSatisfy(result -> {
+                            assertThat(result.headers().lastHeader(Token.HEADER).value())
+                                    .isEqualTo(ContractTestData.REQUEST_TOKEN.getBytes());
+
+                            assertThat(result.value())
+                                    .isEqualTo(ContractTestData.REQUEST_OBJ);
+                        })
+        );
+    }
+
```

Comment:
Add an exception test, what happens if the user doesnt exist?

Reply:
Added exception test and tidied up ContractTestData.